### PR TITLE
feat: implement mail service

### PR DESCRIPTION
**Pull request by khuongln-1346**
# Pull Request types

- ✨ New Feature

# Description
implement mail service

# Checklist

- [x] 1. Build successfully on local
- [x] 2. Format all code files included in the PR
- [x] 3. No more unconfirmed specs in the PR
- [x] 4. Tested the normal case and error cases following specs
- [x] 5. Is there a refactor to the old code? If yes, have you tested the old function and noted the impact yet?
- [ ] 6. Loops have a set length and correct termination conditions
- [ ] 7. There aren't any redundant or duplicate codes in PR (clean code)
- [ ] 8. There aren't any hard-coded in the code in PR
- [ ] 9. Optimization logic handling and optimizations SQL (for example, don't get redundant data or add an index to optimize performance, query only necessary fields...)
- [ ] 10. Update issue status and add PR link to the issue

# Evidence
(Screenshot or Video)
image

# Discussion

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -28,6 +28,15 @@ services:
       - "25432:5432"
     networks:
       - nestjs-boilerplate-network
+
+  maildev:
+    build:
+      context: .
+      dockerfile: maildev.Dockerfile
```
Comment (ngocp-0847) [inline docker-compose.yml:35]: sao ko dùng bản gốc e hè :hehe: https://github.com/maildev/maildev/blob/master/docker-compose.yml

File: src/api/auth/auth.service.ts [hunk 2]
```diff
@@ -84,8 +86,10 @@ export class AuthService {
     });
   }
 
-  async register(_dto: RegisterReqDto): Promise<RegisterResDto> {
-    throw new Error('Method not implemented.');
+  async register(dto: RegisterReqDto): Promise<RegisterResDto> {
+    this.mailService.sendEmailVerification(dto.email, 'test'); // TODO: Update logic when sending email verification
```
Comment (ngocp-0847) [inline src/api/auth/auth.service.ts:90]: 💯 , tích hợp dùng worker luôn e hè, kiểu gì cũng phải có..

File: docker-compose.yml (hunk 1 above)
Comment (khuongln-1346) [reply docker-compose.yml:35]: @ngocp-0847 Bản gốc bị outdate á anh, 2 năm rồi nó chưa update :D

File: src/api/auth/auth.service.ts (hunk 2 above)
Comment (khuongln-1346) [reply src/api/auth/auth.service.ts:90]: @ngocp-0847 À đợi những PR chia ae cùng làm a ợ :v
PR này TO quá rồi

**Pull request merged**